### PR TITLE
Feat/webusb

### PR DESCRIPTION
**Pull request by amanCypherock**
https://app.clickup.com/t/9002019994/CHI-4522

### CODE REVIEW

**Comment by rohanagarwal94** [comment]
Task linked: [CHI-4522 WebUSB Firmware Enabler](https://app.clickup.com/t/9002019994/CHI-4522)

File: common/interfaces/desktop_app_interface/usb_api.c [hunk 1]
```diff
@@ -515,9 +525,22 @@ static void comm_packet_parser(const uint8_t *data, const uint16_t length) {
...
 #else
-    return lusb_write(SDK_RESP_PACKET, sizeof(SDK_RESP_PACKET));
+    return lusb_write(SDK_RESP_PACKET, sizeof(SDK_RESP_PACKET), interface);
 #endif
 
+  if (rx_packet.interface == COMM_LIBUSB__UNDEFINED) {
+    // if(interface != comm_status.active_interface){
+    //   LOG_CRITICAL("USB Conflict 1, iface present:%d, iface new:%d",
+    //   rx_packet.interface, interface); return;
+    // }
```
[the remark this answers is not in the document]
Comment (amanCypherock) [reply common/interfaces/desktop_app_interface/usb_api.c:535]: removed

File: common/interfaces/desktop_app_interface/usb_api.c [hunk 2]
```diff
@@ -515,9 +525,22 @@ static void comm_packet_parser(const uint8_t *data, const uint16_t length) {
...
+    rx_packet.interface = interface;
+  } else if (interface != rx_packet.interface) {
+    LOG_CRITICAL("USB Conflict 2, iface present:%d, iface new:%d",
```
[the remark this answers is not in the document]
Comment (amanCypherock) [reply common/interfaces/desktop_app_interface/usb_api.c:538]: We would never reach this condition as we will be reseting the interface variable before exiting the function

[the remark this answers is not in the document]
Comment (amanCypherock) [reply common/interfaces/desktop_app_interface/usb_api.c:538]: Removing the check

**Pull request merged**